app/scripts/modules/amazon/src/serverGroup/details/amazonServerGroupDetailsGetter.ts

```diff
@@ -0,0 +1,89 @@
+import { IPromise } from 'angular';
+import { isEmpty } from 'lodash';
+import { Observable } from 'rxjs';
+
+import {
+  IServerGroupDetailsProps,
+  IServerGroupDetailsState,
+  ISubnet,
+  ReactInjector,
+} from '@spinnaker/core';
+
+
+import { AwsReactInjector } from 'amazon/reactShims';
+import { IAmazonLoadBalancer, IAmazonServerGroup } from 'amazon/domain';
+
+
+function extractServerGroupSummary(props: IServerGroupDetailsProps): IPromise<IAmazonServerGroup> {
+  const { app, serverGroup } = props;
+  return app.ready()
+    .then(() => {
+      let summary: IAmazonServerGroup = app.serverGroups.data.find((toCheck: IAmazonServerGroup) => {
+        return toCheck.name === serverGroup.name && toCheck.account === serverGroup.accountId && toCheck.region === serverGroup.region;
+      });
+      if (!summary) {
+        app.loadBalancers.data.some((loadBalancer: IAmazonLoadBalancer) => {
+          if (loadBalancer.account === serverGroup.accountId && loadBalancer.region === serverGroup.region) {
+            return loadBalancer.serverGroups.some((possibleServerGroup) => {
+              if (possibleServerGroup.name === serverGroup.name) {
+                summary = possibleServerGroup;
+                return true;
+              }
+              return false;
+            });
+          }
+          return false;
+        });
+      }
+      return summary;
+    }
+  );
+}
+
+export function amazonServerGroupDetailsGetter(props: IServerGroupDetailsProps, newStateCallback: (state: IServerGroupDetailsState) => void, autoClose: () => void): Observable<IAmazonServerGroup> {
```

Comment:
I think we can improve this using RxJS "lettable" operators (now known as "pipeable" operators)

Basically you return an operator that takes an `Observable<T>` and returns an `Observable<R>` where `T` is `IServerGroup` and `R` is `IAmazonServerGroup`